src/System.Globalization/tests/CultureInfo/CultureInfoTest.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Globalization;
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class CultureInfoTest
+    {
+        [Fact]
+        public void TestSettingThreadCultures()
+        {
+            CultureInfo culture = CultureInfo.CurrentCulture;
+            CultureInfo uiCulture = CultureInfo.CurrentUICulture;
+
+            CultureInfo jaCulture = new CultureInfo("ja-JP");
+
+            CultureInfo.CurrentCulture = jaCulture;
+            CultureInfo.CurrentUICulture = jaCulture;
+
+            Assert.True(CultureInfo.CurrentCulture.Equals(jaCulture));
+            Assert.True(CultureInfo.CurrentUICulture.Equals(jaCulture));
+
+            DateTime dt = new DateTime(2014, 3, 14, 3, 14, 0);
+
+            Assert.True(dt.ToString().Equals(dt.ToString(jaCulture)));
+
+            CultureInfo.CurrentCulture = culture;
+            CultureInfo.CurrentUICulture = uiCulture;
```

Comment:
This restoring of the original culture should be done inside a finally block; otherwise if one of the above asserts fails and throws, the wrong culture could be left set for the execution of other tests.